plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -246,7 +259,8 @@ private AzureStorageSettings(
         ProxySettings proxySettings
     ) {
         this.account = account;
-        this.connectString = buildConnectString(account, key, sasToken, endpointSuffix);
+        this.connectString = buildConnectString(account, key, sasToken, endpointSuffix, tokenCredentialType);
```

Comment:
So here we now have 2 strategies:
 1. if `managed` identity is used, the connection string is not relevant
 2. if `managed` identity is not used, the connection string is relevant

We should make that not `if` driven but use endpoint / client build strategy, for example:

```
 private static BlobServiceClientBuilder createClientBuilder(AzureStorageSettings settings) throws InvalidKeyException,
        URISyntaxException {
        if (settings.usesManagedIdentityCredential()) {
            return SocketAccess.doPrivilegedException(
                () -> new BlobServiceClientBuilder().credential(new ManagedIdentityCredentialBuilder().build())
            );
        }
        return SocketAccess.doPrivilegedException(() -> new BlobServiceClientBuilder().connectionString(settings.getConnectString()));
    }
```

Becomes:

```
 private static BlobServiceClientBuilder createClientBuilder(AzureStorageSettings settings) throws InvalidKeyException,
        URISyntaxException {
        return SocketAccess.doPrivilegedException(() -> settings.configure(new BlobServiceClientBuilder()).build());
    }

```

Reply:
refactored